### ISSUE TITLE
prefer_declaring_const_constructors lint showing for the constructors with a super initializer that is not a const

### ISSUE THREAD

**Issue by imsamgarg** [charlescyt/pyramid_lint#20]
### Is there an existing issue for this bug?

- [X] I have searched the [existing issues](https://github.com/charlescyt/pyramid_lint/issues).

### Bug Description

prefer_declaring_const_constructors lint showing for the constructors with a super initializer that is not a const. 

![Screenshot 2024-01-12 211746](https://github.com/charlescyt/pyramid_lint/assets/34547038/db1e0f3d-8e7d-456d-bc12-e7365a74403c)

### Steps to Reproduce

Just create any constructor in any class that has a super initializer that is not const.

**Comment by charlescyt** [comment]
That's because the constructor of `Temp` is not a const constructor.

There are a few edge cases that I didn't consider when I implemented the lint. I will spend some time improving it.
Thanks for raising the issue!

**Comment by charlescyt** [comment]
The lint should not be triggered when the redirecting/super constructor is not a const constructor.
```dart
class Point {
  final double x;
  final double y;

  Point(this.x, this.y);

  Point.origin() : this(0, 0); // Redirecting constructor is not const
}

class Point3D extends Point {
  final double z;

  Point3D(double x, double y, this.z) : super(x, y); // Super constructor is not const

  // Point3D(super.x, super.y, this.z); // Need to handle super parameters as well

  Point3D.origin() // Super constructor is not const
      : z = 0,
        super.origin();
}
```

**Comment by imsamgarg** [comment]
Now i'm getting error on const constructors too
![image](https://github.com/charlescyt/pyramid_lint/assets/34547038/83001d8d-56c8-4de5-864f-67564e032fff)

**Comment by imsamgarg** [comment]
It's also showing all over the place
![image](https://github.com/charlescyt/pyramid_lint/assets/34547038/a874b2d2-e6e6-40da-a90e-2fc675816a09)

**Comment by charlescyt** [comment]
Hello! 
Is `NativeFileType.fromPath` a factory constructor?
Could you provide a minimum reproducible example for me to work on?

**Comment by imsamgarg** [comment]
For some reason it's not showing anymore even I didn't change anything.
